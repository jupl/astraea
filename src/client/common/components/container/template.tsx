import React, {ReactNode} from 'react'

interface Props {
  children: ReactNode
}

/**
 * Render container component
 * @return Decorated component
 */
export const Container = ({children}: Props) => <>{children}</>
