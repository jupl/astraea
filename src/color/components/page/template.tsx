import * as React from 'react'
import styled from 'styled-components'

// Background component
interface BackgroundProps extends React.HTMLProps<HTMLDivElement> {
  color: string
}
const StyledBackground = styled(Background)`
  transition: background 0.8s ease-out;
  display: flex;
  background-color: ${({color}) => color};
`
function Background({color: _, ...props}: BackgroundProps) {
  return <div {...props} />
}

// Text component
const Text = styled.span`
  margin-left: 4px;
  margin-right: 4px;
`

// Gradient overlay component
const Gradient = styled.div`
  flex: 1;
  display: flex;
  justify-content: center;
  align-items: center;
  background: linear-gradient(rgba(255, 255, 255, 0.4), transparent);
`

/** Button component */
export const Button = styled.button`
  border-radius: 50%;
  background: radial-gradient(white, gainsboro);
  border-color: gainsboro;
  outline: none;
`

/** Data for color page component */
export interface Props {
  className?: string
  selectedColor: string
  nextColor(): void
  previousColor(): void
}

/**
 * Render color page component showing current color and buttons to change
 * @param props Component properties
 * @return Color page component
 */
export function ColorPage({
  className,
  nextColor,
  previousColor,
  selectedColor,
}: Props) {
  return (
    <StyledBackground color={selectedColor} className={className}>
      <Gradient>
        <Button onClick={previousColor}>&lt;</Button>
        <Text>Hello, World</Text>
        <Button onClick={nextColor}>&gt;</Button>
      </Gradient>
    </StyledBackground>
  )
}
