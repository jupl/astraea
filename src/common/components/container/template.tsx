import React, {ReactNode} from 'react'

/** Component properties */
export interface Props {
  children: ReactNode
}

/**
 * Render container component
 * @return Decorated component
 */
export const Container = ({children}: Props) => <>{children}</>
