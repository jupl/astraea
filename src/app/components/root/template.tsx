import React from 'react'
import styled from 'styled-components'
import {ColorPage} from '../../../color/components/page'

// Container component
const Container = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  overflow: auto;
  display: flex;
  background-color: white;
`

// Styled color page component
const StyledColorPage = styled(ColorPage)`
  flex: 1;
`

/**
 * Render root component representing the entire application
 * @return Root component
 */
export const AppRoot = () => (
  <Container>
    <StyledColorPage />
  </Container>
)
