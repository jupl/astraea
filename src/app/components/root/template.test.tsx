import {shallow} from 'enzyme'
import React from 'react'
import {AppRoot} from './template'

describe('<AppRoot> Template', () => {
  it('should render as expected', () => {
    expect(shallow(<AppRoot />)).toMatchSnapshot()
  })
})
