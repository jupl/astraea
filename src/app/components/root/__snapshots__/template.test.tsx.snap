// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRoot> Template should render as expected 1`] = `
<styled.div>
  <Styled(Component) />
</styled.div>
`;
